platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/GeoJsonSource.java

```diff
@@ -285,14 +285,20 @@ public void setGeoJson(Geometry geometry) {
    * Updates the GeoJson. The update is performed asynchronously,
    * so the data won't be immediately visible or available to query when this method returns.
    *
-   * @param features the GeoJSON FeatureCollection
+   * @param featureCollection the GeoJSON FeatureCollection
    */
-  public void setGeoJson(FeatureCollection features) {
+  public void setGeoJson(FeatureCollection featureCollection) {
     if (detached) {
       return;
     }
     checkThread();
-    nativeSetFeatureCollection(features);
+
+    List<Feature> features = featureCollection.features();
+    if (features != null) {
+      List<Feature> featuresCopy = new ArrayList<>(features);
+      nativeSetFeatureCollection(FeatureCollection.fromFeatures(featuresCopy));
+    }
+    nativeSetFeatureCollection(featureCollection);
```

Comment:
we are calling `nativeSetFeatureCollection` twice if `featureCollection.features()`is not null

Reply:
🤦‍♂  good catch